CDTDatastore/query/CDTQIndex.m

```diff
@@ -52,22 +52,26 @@ + (NSArray *)validSettings
 
 - (instancetype)initWithFields:(NSArray *)fieldNames
                      indexName:(NSString *)indexName
-                     indexType:(NSString *)indexType
+                     indexType:(CDTQIndexType)indexType
                  indexSettings:(NSDictionary *)indexSettings
 {
     self = [super init];
     if (self) {
         _fieldNames = fieldNames;
         _indexName = indexName;
-        _indexType = indexType;
+        _indexType = [CDTQIndexManager stringForIndexType:indexType];
```

Comment:
I see you override the setters to write both fields. Personally, I think you should instead override the getter/setter for one of the properties -- preferably the string one -- so you're able to have just a single underlying instance variable -- the CDTQIndexType one.